src/openloco/TrackData.cpp

```diff
@@ -0,0 +1,307 @@
+    const std::vector<PreviewTrack> trackPiece43 = {
+        PreviewTrack{ 0, 0, 0, 0, 0, 3, 0 },
+    };
+
+    // 0x004F73D8
+    const std::vector<std::vector<PreviewTrack>> trackPieces = {
+        trackPiece0,
+        trackPiece1,
+        trackPiece2,
+        trackPiece3,
+        trackPiece4,
+        trackPiece5,
+        trackPiece6,
+        trackPiece7,
+        trackPiece8,
+        trackPiece9,
+        trackPiece10,
+        trackPiece11,
+        trackPiece12,
+        trackPiece13,
+        trackPiece14,
+        trackPiece15,
+        trackPiece16,
+        trackPiece17,
+        trackPiece18,
+        trackPiece19,
+        trackPiece20,
+        trackPiece21,
+        trackPiece22,
+        trackPiece23,
+        trackPiece24,
+        trackPiece25,
+        trackPiece26,
+        trackPiece27,
+        trackPiece28,
+        trackPiece29,
+        trackPiece30,
+        trackPiece31,
+        trackPiece32,
+        trackPiece33,
+        trackPiece34,
+        trackPiece35,
+        trackPiece36,
+        trackPiece37,
+        trackPiece38,
+        trackPiece39,
+        trackPiece40,
+        trackPiece41,
+        trackPiece42,
+        trackPiece43,
+    };
+
+    const std::vector<PreviewTrack> roadPiece0 = {
+        PreviewTrack{ 0, 0, 0, 0, 0, 15, 0 },
+    };
+    const std::vector<PreviewTrack> roadPiece1 = {
+        PreviewTrack{ 0, 0, 0, 0, 0, 15, 0 },
+    };
+    const std::vector<PreviewTrack> roadPiece2 = {
+        PreviewTrack{ 0, 0, 0, 0, 0, 15, 0 },
+    };
+    const std::vector<PreviewTrack> roadPiece3 = {
+        PreviewTrack{ 0, 0, 0, 0, 0, 7, 0 },
+        PreviewTrack{ 1, 0, -32, 0, 0, 8, PreviewTrackFlags::diagonal },
+        PreviewTrack{ 2, -32, 0, 0, 0, 2, 0 },
+        PreviewTrack{ 3, -32, -32, 0, 0, 7, 0 },
+    };
+    const std::vector<PreviewTrack> roadPiece4 = {
+        PreviewTrack{ 0, 0, 0, 0, 0, 11, 0 },
+        PreviewTrack{ 1, 0, 32, 0, 0, 4, PreviewTrackFlags::diagonal },
+        PreviewTrack{ 2, -32, 0, 0, 0, 1, 0 },
+        PreviewTrack{ 3, -32, 32, 0, 0, 11, 0 },
+    };
+    const std::vector<PreviewTrack> roadPiece5 = {
+        PreviewTrack{ 0, 0, 0, 0, 16, 15, PreviewTrackFlags::unk4 },
+        PreviewTrack{ 1, -32, 0, 0, 16, 207, PreviewTrackFlags::unk4 },
+    };
+    const std::vector<PreviewTrack> roadPiece6 = {
+        PreviewTrack{ 0, 0, 0, -16, 16, 63, PreviewTrackFlags::unk4 },
+        PreviewTrack{ 1, -32, 0, -16, 16, 15, PreviewTrackFlags::unk4 },
+    };
+    const std::vector<PreviewTrack> roadPiece7 = {
+        PreviewTrack{ 0, 0, 0, 0, 16, 207, PreviewTrackFlags::unk3 | PreviewTrackFlags::unk2 },
+    };
+    const std::vector<PreviewTrack> roadPiece8 = {
+        PreviewTrack{ 0, 0, 0, -16, 16, 63, PreviewTrackFlags::unk1 | PreviewTrackFlags::unk0 },
+    };
+    const std::vector<PreviewTrack> roadPiece9 = {
+        PreviewTrack{ 0, 0, 0, 0, 0, 3, 0 },
+    };
+
+    // 0x004F6D1C
+    const std::vector<std::vector<PreviewTrack>> roadPieces = {
```

Comment:
this could become a `std::array<std::vector>`. Its not really going to make much difference though but further enforces the fixed size.